samples_project/Assets/SampleViewer/Samples/Measure/Measure.cs

```diff
@@ -14,272 +14,223 @@
+				stops.Push(lineMarker);
+				RenderLine(ref featurePoints);
+				RebaseLine();
+
+			}
+
+		}
+	}
+
+	private void Interpolate(GameObject start, GameObject end, List<GameObject> featurePoints)
+	{
+		var startPoint = start.GetComponent<ArcGISLocationComponent>().Position;
+		var endPoint = end.GetComponent<ArcGISLocationComponent>().Position;
+
+		double d = ArcGISGeometryEngine.DistanceGeodetic(startPoint, endPoint, new ArcGISLinearUnit(ArcGISLinearUnitId.Meters), new ArcGISAngularUnit(ArcGISAngularUnitId.Degrees), ArcGISGeodeticCurveType.Geodesic).Distance;
+		float n = Mathf.Floor((float)d / InterpolationInterval);
+		double dx = (end.transform.position.x - start.transform.position.x) / n;
+		double dz = (end.transform.position.z - start.transform.position.z) / n;
+
+		var pre = start.transform.position;
+
+		//calculate n-1 intepolation points/n-1 segments because the last segment is already created by the end point 
+		for (int i = 0; i < n - 1; i++)
+		{
+			GameObject next = Instantiate(InterpolationMarker, arcGISMapComponent.transform);
+
+			//calculate transform of next point
+			float nextX = pre.x + (float)dx;
+			float nextZ = pre.z + (float)dz;
+			next.transform.position = new Vector3(nextX, 0, nextZ);
+
+			//set default location component of next point
+			next.GetComponent<ArcGISLocationComponent>().enabled = true;
+			next.GetComponent<ArcGISLocationComponent>().Rotation = new ArcGISRotation(0, 90, 0);
+
+			//define height
+			SetElevation(next);
+
+			featurePoints.Add(next);
+
+			pre = next.transform.position;
+		}
+
+	}
+
+	// set height for point transform and location component
+	void SetElevation(GameObject stop)
+	{
+		// start the raycast in the air at an arbitrary to ensure it is above the ground
+		var raycastHeight = 5000;
+		var position = stop.transform.position;
+		var raycastStart = new Vector3(position.x, position.y + raycastHeight, position.z);
+		if (Physics.Raycast(raycastStart, Vector3.down, out RaycastHit hitInfo))
+		{
+			var location = stop.GetComponent<ArcGISLocationComponent>();
+			location.Position = arcGISMapComponent.EngineToGeographic(hitInfo.point);
+			stop.transform.position = hitInfo.point;
+		}
+	}
+
+	private void RenderLine(ref List<GameObject> featurePoints)
+	{
+
+		lineRenderer.widthMultiplier = 5;
+
+		var allPoints = new List<Vector3>();
+
+		foreach (var stop in featurePoints)
+		{
+			if (stop.transform.position.Equals(Vector3.zero))
+			{
+				Destroy(stop);
+				continue;
+			}
+			allPoints.Add(stop.transform.position);
+		}
+
+		lineRenderer.positionCount = allPoints.Count;
+		lineRenderer.SetPositions(allPoints.ToArray());
+	}
+
+	public void ClearLine()
+	{
+		foreach (var stop in featurePoints)
+			Destroy(stop);
+		featurePoints.Clear();
+		stops.Clear();
+		geodeticDistance = 0;
+		GeodeticDistanceText.text = "Distance: " + geodeticDistance + unitTxt;
```

Comment:
I suggest using `UnitChanged()` method could be called to always show the geodetic distance in selected units.
Is it necessary to do all of the math in code in the selected unit as well?